examples/imgui_impl_x11.cpp

```diff
@@ -0,0 +1,215 @@
+    io.KeyMap[ImGuiKey_RightArrow] = XK_Right - 0xFF00;
+    io.KeyMap[ImGuiKey_UpArrow] = XK_Up - 0xFF00;
+    io.KeyMap[ImGuiKey_DownArrow] = XK_Down - 0xFF00;
+    io.KeyMap[ImGuiKey_PageUp] = XK_Page_Up - 0xFF00;
+    io.KeyMap[ImGuiKey_PageDown] = XK_Page_Up - 0xFF00;
+    io.KeyMap[ImGuiKey_Home] = XK_Home - 0xFF00;
+    io.KeyMap[ImGuiKey_End] = XK_End - 0xFF00;
+    io.KeyMap[ImGuiKey_Insert] = XK_Insert - 0xFF00;
+    io.KeyMap[ImGuiKey_Delete] = XK_Delete - 0xFF00;
+    io.KeyMap[ImGuiKey_Backspace] = XK_BackSpace - 0xFF00;
+    io.KeyMap[ImGuiKey_Space] = XK_space;
+    io.KeyMap[ImGuiKey_Enter] = XK_Return - 0xFF00;
+    io.KeyMap[ImGuiKey_Escape] = XK_Escape - 0xFF00;
+    io.KeyMap[ImGuiKey_KeyPadEnter] = XK_KP_Enter - 0xFF00;
+    io.KeyMap[ImGuiKey_A] = 'A';
+    io.KeyMap[ImGuiKey_C] = 'C';
+    io.KeyMap[ImGuiKey_V] = 'V';
+    io.KeyMap[ImGuiKey_X] = 'X';
+    io.KeyMap[ImGuiKey_Y] = 'Y';
+    io.KeyMap[ImGuiKey_Z] = 'Z';
+
+    return true;
+}
+
+void    ImGui_ImplX11_Shutdown()
+{
+    xcb_key_symbols_free(g_KeySyms);
+    xcb_flush(g_Connection);
+    xcb_disconnect(g_Connection);
+}
+
+void    ImGui_ImplX11_NewFrame()
+{
+    ImGuiIO& io = ImGui::GetIO();
+    IM_ASSERT(io.Fonts->IsBuilt() && "Font atlas not built! It is generally built by the renderer back-end. Missing call to renderer _NewFrame() function? e.g. ImGui_ImplOpenGL3_NewFrame().");
+
+    // Setup time step
+    clock_gettime(CLOCK_MONOTONIC_RAW, &g_CurrentTime);
+    io.DeltaTime = (g_CurrentTime.tv_sec - g_LastTime.tv_sec) +
+        ((g_CurrentTime.tv_nsec - g_LastTime.tv_nsec) / 1000000000.0f);
+    g_LastTime = g_CurrentTime;
+}
+
+// X11 xcb message handler (process X11 mouse/keyboard inputs, etc.)
+// Call from your application's message handler. Returns true if
+// ImGui processed the event. You can use this for checking if you
+// need further processing for the event.
+bool ImGui_ImplX11_Event(xcb_generic_event_t *event)
+{
+    if (ImGui::GetCurrentContext() == NULL)
+        return false;
+
+    ImGuiIO& io = ImGui::GetIO();
+    switch (event->response_type & ~0x80) {
+    case XCB_MOTION_NOTIFY: {
+        xcb_motion_notify_event_t *e = (xcb_motion_notify_event_t*)event;
+        io.MousePos = ImVec2(e->event_x, e->event_y);
+        return true;
+    }
+    case XCB_KEY_PRESS: {
+        xcb_key_press_event_t *e = (xcb_key_press_event_t *)event;
+        xcb_keysym_t k = xcb_key_press_lookup_keysym(g_KeySyms, e, 0);
+        if(k < 0xFF) { // latin-1 range
+            io.AddInputCharacter(k);
+            io.KeysDown[k] = 1;
+        }
+        else if(k >= XK_Shift_L && k <= XK_Hyper_R) { // modifier keys
+            switch(k) {
+                case XK_Shift_L:
+                case XK_Shift_R:
+                    io.KeyShift = true;
+                    break;
+                case XK_Control_L:
+                case XK_Control_R:
+                    io.KeyCtrl = true;
+                    break;
+                case XK_Meta_L:
```

Comment:
The technical definition is that the meta and alt keys are different, in practice I don't think that's actually true in modern times.